auth/standard_role_manager.cc

```diff
@@ -701,4 +714,137 @@ future<> standard_role_manager::remove_attribute(std::string_view role_name, std
+                get_auth_ks_name(_qp),
+                meta::roles_table::name);
+    });
+
+    const auto roles = co_await _qp.execute_internal(
+            role_query,
+            db::consistency_level::LOCAL_ONE, // TODO: Verify later if this is okay.
+            internal_distributed_query_state(),
+            cql3::query_processor::cache_internal::yes);
+
+    if (roles->empty()) [[unlikely]] {
+        on_internal_error(log, "No role has been found in the system");
+    }
+
+    std::vector<role_info> result{};
+    result.reserve(roles->size());
+
+    for (const auto& role_record : *roles) {
+        role_info ri{};
+
+        ri.role_name = role_record.get_as<sstring>(meta::roles_table::role_col_name);
+        ri.config.can_login = role_record.get_or<bool>(meta::roles_table::can_login_col_name, false);
+        ri.config.is_superuser = role_record.get_or<bool>(meta::roles_table::is_superuser_col_name, false);
+
+        if (with_salted_hashes && role_record.has(meta::roles_table::salted_hash_col_name)) {
+            ri.maybe_salted_hash = role_record.get_as<sstring>(meta::roles_table::salted_hash_col_name);
+        }
+
+        result.push_back(std::move(ri));
+        co_await coroutine::maybe_yield();
+    }
+
+    co_return result;
 }
+
+future<std::vector<role_grants>> standard_role_manager::get_role_grants() const {
+    // We don't use the `role_members` table here because it keeps track of this data
+    // in a "dense" way, i.e. pairs (granted role, grantee), so we might end up browsing
+    // O(n^2) records where n is the number of roles. Using the `roles` table, we are
+    // guaranteed to get at most O(n) rows.
+    const auto grants_query = seastar::format("SELECT {}, {} FROM {}.{}",
+            meta::roles_table::role_col_name,
+            meta::roles_table::member_of_col_name,
+            get_auth_ks_name(_qp),
+            meta::roles_table::name,
+            meta::roles_table::member_of_col_name);
+
+    const auto grants = co_await _qp.execute_internal(
+            grants_query,
+            db::consistency_level::LOCAL_ONE, // TODO: Verify later if this is okay.
+            internal_distributed_query_state(),
+            cql3::query_processor::cache_internal::yes);
+
+    std::vector<role_grants> result{};
+    result.reserve(grants->size());
+
+    for (const auto& role_record : *grants) {
+        // Unfortunately, Scylla doesn't allow us to do `WHERE member_of IS NOT NULL`,
+        // so we need to filter out the relevant results by hand.
+        if (!role_record.has(meta::roles_table::member_of_col_name)) {
+            continue;
+        }
+
+        role_grants rg{};
+
+        rg.grantee_role = role_record.get_as<sstring>(meta::roles_table::role_col_name);
+        rg.granted_roles = role_record.get_list<sstring>(meta::roles_table::member_of_col_name);
+
+        result.push_back(std::move(rg));
+        co_await coroutine::maybe_yield();
+    }
+
+    result.shrink_to_fit();
```

Comment:
what this achieves? if size is known upfront

Reply:
The size isn't known at first. In the loop above we need to filter the roles that inherit from at least one other role. If there are few of such roles, the vector will be small. But you're right; perhaps a better approach would be to ditch both the `reserve` and `shrink_to_fit` calls

Reply:
if you don't need to optimize (seems to be the case here) it's simpler not to